pkg/controller/oneagent/oneagent_controller.go

```diff
@@ -290,7 +294,7 @@ func (r *ReconcileOneAgent) reconcileRollout(logger logr.Logger, instance *dynat
 		desired, err := dtc.GetLatestAgentVersion(dtclient.OsUnix, dtclient.InstallerTypeDefault)
 		if err != nil {
 			logger.Error(err, "failed to get desired version")
-			return updateCR, nil
+			return updateCR, err
```

Comment:
Just logging and not returning an error was on purpose here I think, since we can do some work even if we can't access the API.